File: website/src/app/models/city.ts
import { Country } from "./country";
import { ToDo } from "./todo";

export interface City {
    id:number;
    name : string;
    country : Country;
    countryId:number;
    sortOrder: number;
    startDate: Date;
    endDate: Date;
    toDos: ToDo[];
}

export interface CityDialogData {
    countryId: number;
    name: string;
    sortOrder:number;
  }

  export interface UpdateCityDialogData {
    id: number;
    name: string;
    startDate: Date| undefined;
    endDate: Date | undefined;
    countryId: number;
    sortOrder: number;
  }

File: Api/Models/City.cs
using System.ComponentModel;

namespace TravelPlannerApp.Models
{
    public class City
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public Country Country{ get; set; }
        public List<ToDo> ToDos { get; set; }
        public DateOnly? StartDate { get; set; }
        public DateOnly? EndDate { get; set; }
        [DefaultValue(0)]
        public int SortOrder { get; set; } = 0;
    }
}
